docs/contracts-intro.md

```diff
@@ -2,11 +2,245 @@
 
 ---8<-- 'includes/data-builder-danger-header.md'
 
-This page will introduce OpenSAFELY Contracts, explaining:
+## OpenSAFELY Contracts are a clinical data specification
 
-* what they are
-* the process by which new contracts could be proposed
-* where [details of existing contracts will be
-  found](contracts-reference.md)
+An *OpenSAFELY Contract* is a core concept in the OpenSAFELY framework.
+Anyone using Data Builder should familiarise themselves with how
+Contracts work.
+
+This page describes:
+
+* what OpenSAFELY Contracts are
+* how OpenSAFELY Contracts relate to dataset definitions
+
+Each OpenSAFELY Contract is a formal specification relating to a
+particular clinical data domain.
+
+!!! note
+    For researchers, the human-readable version of each Contract is
+    listed in the [Contracts reference](contracts-reference.md).
+
+    For developers working on OpenSAFELY integrations, you can also see
+    how [each Contract is specified in Python
+    code](https://github.com/opensafely-core/databuilder/blob/main/databuilder/contracts/).
+    **It is not necessary to understand the Contract's Python code to be
+    able to use a Contract.**
+
+OpenSAFELY Contracts are authored by the OpenSAFELY platform developers:
+the Contract designs are informed by discussion and feedback from [data
+providers](data-provider-integration.md) and researchers.
+
+### How do OpenSAFELY Contracts fit into OpenSAFELY?
+
+Researchers write [dataset definitions](dataset-definition.md) to access
+data tables from OpenSAFELY backends. This is a simple example of a
+dataset definition that accesses a data table:
```

Comment:
I think we should explain which _bit_ accesses the data table - i.e. in this case, it's `patients`.

Possibly we should just show these two lines

```py
from databuilder.tables import patients
year_of_birth = patients.date_of_birth.year
```

And explain that it accesses the `date_of_birth` field of the `patients` table?